### PR TITLE
Add `useUpdateThermostat`

### PR DESCRIPTION
**Pull request by xplato**

### CODE REVIEW

**Comment by xplato** [comment]
@razor-x do we have any good ways of testing this outside of components?

**Comment by razor-x** [comment]
This is setup: https://testing-library.com/

What do you want to test? If you want to test the hook I think it has to be tested inside a component. Maybe testing library has a pattern to get around that though

**Comment by xplato** [comment]
@razor-x yeah, just testing the hook. Can this be done in a separate PR by implementing some mutation features?

File: src/lib/seam/thermostats/use-update-thermostat.ts [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  useQueryClient,
+} from '@tanstack/react-query'
+import type {
+  SeamError,
+  ThermostatDevice,
+  ThermostatsListResponse,
+  ThermostatUpdateRequest,
+} from 'seamapi'
+
+import { NullSeamClientError, useSeamClient } from 'lib/index.js'
```
Comment (razor-x) [inline src/lib/seam/thermostats/use-update-thermostat.ts:13]:
```suggestion
import { NullSeamClientError, useSeamClient } from 'lib/seam/use-seam-client.js'
```

File: src/lib/seam/thermostats/use-update-thermostat.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+          })
+        }
+      )
+    },
+  })
+}
+
+function getUpdatedThermostat(
+  thermostat: ThermostatDevice,
+  variables: UseUpdateThermostatMutationParams
```
Comment (razor-x) [inline src/lib/seam/thermostats/use-update-thermostat.ts:70]: Was this param meant to be used?

Comment (xplato) [reply src/lib/seam/thermostats/use-update-thermostat.ts:71]: `variables`? Yes, it is used below on line 87 (unless I misunderstood)

Comment (razor-x) [reply src/lib/seam/thermostats/use-update-thermostat.ts:71]: Ah ok, my browser search highlighter didn't work right so I missed it.

File: src/lib/seam/thermostats/use-update-thermostat.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+): ThermostatDevice {
+  return {
+    ...thermostat,
+
+    properties: {
+      ...thermostat.properties,
+
+      default_climate_setting: {
+        ...thermostat.properties.default_climate_setting,
+        manual_override_allowed:
```
Comment (razor-x) [inline src/lib/seam/thermostats/use-update-thermostat.ts:81]: Can you explain this logic a bit? Why do we default to false here vs. allow the null? Should we default to `variables.<something>.manual_override_allowed`?

**Review by razor-x — APPROVED** [review]
Just the question on the optimistic update logic. Probably something a test would have explained.

File: src/lib/seam/thermostats/use-update-thermostat.ts (hunk 3 above)
Comment (xplato) [reply src/lib/seam/thermostats/use-update-thermostat.ts:81]: Yeah, I struggled with this for a bit. This is just to make TS happy. The problem is that the type for `manual_override_allowed ` when updating is `boolean | undefined`, but it has to be `boolean`—It won't allow `null`. We can also get around this by writing:

```
 default_climate_setting: {
  ...thermostat.properties.default_climate_setting,
  ...variables.default_climate_setting,
} as ClimateSetting
```

We could default to the definition in `variables`, but that would require an additional null-check and a subsequent default.

Comment (xplato) [reply src/lib/seam/thermostats/use-update-thermostat.ts:81]: I'm going to merge this in to unblock @Andrewjeska—logic can be adjusted if needed during implementation.

Comment (razor-x) [reply src/lib/seam/thermostats/use-update-thermostat.ts:81]: > This is just to make TS happy

I trust the types! If we type cast we may actually introduce an unexpected null.

**Pull request merged**